packages/xrpl/src/models/utils/flags.ts

```diff
@@ -60,34 +58,52 @@ const txToFlag = {
 /**
  * Sets a transaction's flags to its numeric representation.
  *
+ * @deprecated
+ * This utility function is deprecated.
+ * Use convertTxFlagsToNumber() instead and use the returned value to modify the tx.Flags from the caller.
+ *
  * @param tx - A transaction to set its flags to its numeric representation.
  */
 export function setTransactionFlagsToNumber(tx: Transaction): void {
-  if (tx.Flags == null) {
-    tx.Flags = 0
-    return
+  // eslint-disable-next-line no-console -- intended deprecation warning
+  console.warn(
+    'This function is deprecated. Use convertTxFlagsToNumber() instead and use the returned value to modify the tx.Flags from the caller.',
+  )
+
+  if (tx.Flags) {
+    // eslint-disable-next-line no-param-reassign -- intended param reassign in setter
+    tx.Flags = convertTxFlagsToNumber(tx)
   }
+}
+
+/**
+ * Returns a transaction's flags as its numeric representation.
+ *
+ * @param tx - A transaction to parse flags for
+ * @returns A numerical representation of a transaction's flags
+ */
+export function convertTxFlagsToNumber(tx: Transaction): number {
   if (typeof tx.Flags === 'number') {
```

Comment:
```suggestion
  if (tx.Flags != null and typeof tx.Flags === 'number') {
```

Reply:
Not sure if this is necessary, wouldn't typeof null return 'object', which would fail this condition anyways?

Reply:
ah, I wasn't aware of this JS behavior 👍